Makefile

```diff
@@ -122,16 +122,24 @@ unit_test_pkg_binlog: tools_setup
 unit_test_others: tools_setup
 	$(call run_unit_test,$(PACKAGES_OTHERS),unit_test_others)
 
-check: tools_setup fmt lint terror_check tidy_mod
+check: tools_setup lint fmt terror_check tidy_mod
 
-fmt:
-	@echo "gofumports"
-	tools/bin/gofumports -w -d -local $(PACKAGE_NAME) $(PACKAGE_DIRECTORIES) 2>&1 | awk '{print} END{if(NR>0) {exit 1}}'
+fmt: tools_setup
+	@if [[ "${nolint}" != "true" ]]; then\
+		echo "shfmt"; \
+		tools/bin/shfmt -l -w -d "tests/" ; \
+		echo "gofumports"; \
+		tools/bin/gofumports -w -d -local $(PACKAGE_NAME) $(PACKAGE_DIRECTORIES) 2>&1 | awk "{print} END{if(NR>0) {exit 1}}" ;\
+		echo "golangci-lint"; \
+		tools/bin/golangci-lint run --config=$(CURDIR)/.golangci.yml --fix --issues-exit-code=1 $(PACKAGE_DIRECTORIES) ;\
+	fi
```

Comment:
Why fmt contains lint? Now if we execute make build, golangci-lint and shfmt will run multiple times

Reply:
golangci-lint with arg `--fix`  means fix unlint code, so this is fmt not lint any more.

after this change, `make check` will run in this order:
```
...
lint (both golangci-lint and shfmt will just check code and not fix)
fmt ( golangci-lint and shfmt will fix the wrong style code)
...
```